include/perfetto/ext/base/sched.h

```diff
@@ -0,0 +1,225 @@
+  static SchedConfig CreateOther(int nice) {
+    return {SchedPolicy::kOther, 0, nice};
+  }
+  static SchedConfig CreateBatch(int nice) {
+    return {SchedPolicy::kBatch, 0, nice};
+  }
+  static SchedConfig CreateFifo(unsigned int priority) {
+    return {SchedPolicy::kFifo, priority, 0};
+  }
+  static SchedConfig CreateRr(unsigned int priority) {
+    return {SchedPolicy::kRr, priority, 0};
+  }
+
+  SchedConfig(const SchedPolicy policy,
+              const unsigned int priority,
+              const int nice)
+      : policy_(policy), rt_priority_(priority), nice_(nice) {
+    switch (policy) {
+      case SchedPolicy::kIdle:
+        PERFETTO_DCHECK(priority == 0 && nice == 0);
+        break;
+      case SchedPolicy::kOther:
+      case SchedPolicy::kBatch:
+        PERFETTO_DCHECK(ValidateNiceValue(nice).ok() && priority == 0);
+        break;
+      case SchedPolicy::kFifo:
+      case SchedPolicy::kRr:
+        PERFETTO_DCHECK(ValidatePriority(priority).ok() && nice == 0);
+        break;
+    }
+  }
+
+  std::string ToString() const;
+
+  /**
+   * Return true if this priority is _lower_ than others'
+   * (that implies `this.KernelPriority()` is _higher_ than
+   * `other.KernelPriority()`).
+   *
+   * If both priorities have the same `KernelPriority()` value, we compare the
+   * policies.
+   *
+   * Strictly speaking, it is not correct to compare realtime priorities
+   * 'SCHED_FIFO' and 'SCHED_RR' based on their policy, but we do it for
+   * simplicity.
+   */
+  bool operator<(const SchedConfig& other) const {
+    const int inverted_priority =
+        -1 * static_cast<int>(InternalUnifiedPriority());
+    const int inverted_other_priority =
+        -1 * static_cast<int>(other.InternalUnifiedPriority());
+    return std::tie(inverted_priority, policy_) <
+           std::tie(inverted_other_priority, other.policy_);
+  }
+
+  bool operator==(const SchedConfig& other) const {
+    return std::tie(policy_, rt_priority_, nice_) ==
+           std::tie(other.policy_, other.rt_priority_, other.nice_);
+  }
+  bool operator!=(const SchedConfig& other) const { return !(*this == other); }
+
+  SchedPolicy policy() const { return policy_; }
+  unsigned int priority() const { return rt_priority_; }
+  int nice() const { return nice_; }
+
+  /**
+   * Return the kernel priority value, as visible on the 'prio :' line in
+   * `/proc/<pid>/sched` file.
+   *
+   * The lower value means the _higher_ priority.
+   */
+  unsigned int KernelPriority() const {
+    switch (policy_) {
+      case SchedPolicy::kIdle:
+        // For SCHED_IDLE kernel priority is always 120
+        return kKernelDefaultPrio;
+      case SchedPolicy::kOther:
+      case SchedPolicy::kBatch:
+        return static_cast<unsigned int>(kKernelDefaultPrio + nice_);
+      case SchedPolicy::kFifo:
+      case SchedPolicy::kRr:
+        return kKernelMaxRtPrio - 1 - rt_priority_;
+    }
+    PERFETTO_FATAL("Can't be here (Unknown sched policy enum value %d)",
```

Comment:
is this needed? the compiler should realized you are exhausting all the switch cases. just return 0 (Add a dcheck if you want) but don't bloat the binary plz